### PR TITLE
changing global username

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Adding the appropriate IDP at the end of the usernames to write to the DB.

https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2242

## Types of changes
Made changes to the auth service layer to give the appropriate username format to all the routes/controllers.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-247

**Review by loneil — CHANGES_REQUESTED** [review]
Blocking until we discuss the additional changes to user db record happening

**Review by loneil — APPROVED** [review]

File: app/src/forms/auth/service.js [hunk 1]
```diff
@@ -129,8 +130,10 @@ const service = {
       // what if name or email changed?
       user = await service.updateUser(user.id, obj);
     }
+
+    const usernameIdp = user.idpCode ? `${user.username}@${user.idpCode}` : user.username;
     // return with the db id...
-    return { id: user.id, ...userInfo };
+    return { id: user.id, usernameIdp: usernameIdp,  ...userInfo };
```
Comment (jujaga) [inline app/src/forms/auth/service.js:136]: Now that I look at this, may not even need to do that const declaration and in-line the ternary directly as an option:
```js
return {
  id: user.id,
  usernameIdp: user.idpCode ? `${user.username}@${user.idpCode}` : user.username,
  ...userInfo
};
```

**Review by jujaga — APPROVED** [review]
Logically looks pretty good. Just a minor comment to consider. Tentatively approving as I'm not seeing anything too major otherwise. 👍

File: app/src/forms/auth/service.js (hunk 1 above)
Comment (MichaelChung123) [reply app/src/forms/auth/service.js:136]: okay, will change this to be an inline ternary value

File: app/src/docs/v1.api-spec.yaml [hunk 2]
```diff
@@ -1723,6 +1723,11 @@ paths:
           schema:
             type: string
           description: user username
+        - in: query
+          name: usernameIdp
+          schema:
+            type: string
+          description: user username with idp type appended
```
Comment (loneil) [inline app/src/docs/v1.api-spec.yaml:1730]: These ones not needed

**Review by TimCsaky — APPROVED** [review]

**Pull request merged**